--- device.c.orig	2025-01-01 00:00:00.000000000 +0000
+++ device.c	2025-01-01 00:00:00.000000000 +0000
@@ -14,6 +14,8 @@
 #include <linux/module.h>
 #include <linux/rtnetlink.h>
 #include <linux/inet.h>
+#include <linux/version.h>
+#include <net/gso.h>
 #include <linux/netdevice.h>
 #include <linux/inetdevice.h>
 #include <linux/if_arp.h>
@@ -286,7 +288,9 @@
 #else
 	dev->tx_queue_len = 0;
 #endif
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,1,0)
 	dev->features |= NETIF_F_LLTX;
+#endif
 	dev->features |= WG_NETDEV_FEATURES;
 	dev->hw_features |= WG_NETDEV_FEATURES;
 	dev->hw_enc_features |= WG_NETDEV_FEATURES;
--- uapi/wireguard.h.orig	2025-01-01 00:00:00.000000000 +0000
+++ uapi/wireguard.h	2025-01-01 00:00:00.000000000 +0000
@@ -153,7 +153,7 @@
 #ifndef _WG_UAPI_WIREGUARD_H
 #define _WG_UAPI_WIREGUARD_H
 
-#define WG_GENL_NAME "amneziawg"
+#define WG_GENL_NAME amneziawg
 #define WG_GENL_VERSION 1
 
 #define WG_KEY_LEN 32
--- main.c.orig	2025-01-01 00:00:00.000000000 +0000
+++ main.c	2025-01-01 00:00:00.000000000 +0000
@@ -9,7 +9,7 @@
 #include "queueing.h"
 #include "ratelimiter.h"
 #include "netlink.h"
-#include "uapi/wireguard.h"
+#include <uapi/linux/wireguard.h>
 #include "crypto/zinc.h"
 
 #include <linux/init.h>
